contracts/Core/AssetManager.sol

```diff
@@ -64,22 +64,35 @@ contract AssetManager is ACL, AssetStorage {
 
     event CollectionActivityStatus(uint256 id, uint256 epoch, bool active, uint256 timestamp);
 
-    constructor(address parametersAddress) {
+    modifier checkState(uint256 state) {
+        require(state == parameters.getState(), "incorrect state");
+        _;
+    }
+
+    modifier checkStates(uint256 state1, uint256 state2) {
+        require(state1 == parameters.getState() || state2 == parameters.getState(), "incorrect states");
+        _;
+    }
+
+    constructor(address parametersAddress, address blockManagerAddress) {
         parameters = IParameters(parametersAddress);
+        blockManager = IBlockManager(blockManagerAddress);
     }
 
     function createJob(
         string calldata url,
         string calldata selector,
-        string calldata name,
-        bool repeat
+        string calldata name
     ) external onlyRole(parameters.getAssetModifierHash()) {
-        numAssets = numAssets + 1;
         uint256 epoch = parameters.getEpoch();
-        Structs.Job memory job = Structs.Job(numAssets, epoch, url, selector, name, repeat, true, msg.sender, 0, uint256(assetTypes.Job));
-        jobs[numAssets] = job;
+        require(numAssetsCreated[epoch] < parameters.maxAssetsCreatedPerEpoch(), "Asset Creation Limit Reached");
+
+        numAssets = numAssets + 1;
+        numAssetsCreated[epoch]++;
 
-        emit JobCreated(numAssets, epoch, url, selector, name, repeat, msg.sender, block.timestamp, assetTypes.Job);
+        Structs.Job memory job = Structs.Job(numAssets, epoch, url, selector, name, true, msg.sender, uint256(assetTypes.Job));
+        jobs[numAssets] = job;
+        emit JobCreated(numAssets, epoch, url, selector, name, msg.sender, block.timestamp, assetTypes.Job);
     }
 
     function updateJob(
```

Comment:
I think this also we should restrict to dispute and confirm only like we have for activate/deactivate

Reply:
Why exactly? Because creation of jobs does not affect active assets as no staker would report for a job anymore.

Reply:
Lets say we allow update in commit as well.

Job : URL : xxx
Half of commit phase has passed
Job has been updated now 
Job : URL : yyy
Some stakers commit as per this URL

at the end the minority would be penalised for no valid reason

Reply:
Okay yes, makes sense. I thought you were talking about createJob. Alright, noted